public/src/pages/RegisterPage/RegisterPage.js

```diff
@@ -45,96 +46,121 @@ export default class RegisterPage extends BasePage {
                 const element = form.querySelector(`[name=${fields[page].name}]`);
-                data.push(element.value);
+                data[fields[page].name] = element.value;
             });
-            if (data[data.length - 1] !== data[data.length - 2]) {
-                validation.getErrorMessage(document.getElementById(fields.repeatPassword.name),
-                    'repeatPasswordError', 'Введенные пароли не совпадают');
-            } else {
-                if (document.getElementById('repeatPasswordError') !== null) {
-                    document.getElementById('repeatPasswordError').remove();
-                }
-            }
 
             //  timing email
-            data[1] = data[1].trim();
-            const [username, email, password, anotherPassword] = data;
-
-            if (validation.validateRegFields(email, password, anotherPassword)) {
-                const r = new Req();
-                const [status] = await r.makePostRequest('api/v1/signup',
-                    {password, email, username});
-
-                switch (status) {
-                case 201:
-                    console.log('auth');
-                    config.authorised = true;
-                    form.removeEventListener('focusout', realTimeCheckHandler);
-                    form.removeEventListener('submit', onSubmitHandler);
-                    config.header.main.render(config);
-                    break;
-                case 400:
-                    document.getElementById('Error400Message') === null ?
-                        validation.getServerMessage(document.getElementById('inForm'),
-                            'Error400Message', 'Ошибка. Попробуйте еще раз') :
-                        console.log('bad request: ', status);
-                    break;
-                case 409:
-                    validation.getErrorMessage(document.getElementById(fields.email.name),
-                        'emailError', 'Почта уже занята');
-                    console.log('no auth: ', status);
-                    break;
-                default:
-                    document.getElementById('serverErrorMessage') === null ?
-                        validation.getServerMessage(document.getElementById('inForm'),
-                            'serverErrorMessage', 'Ошибка сервера. Попробуйте позже') :
-                        console.log('server error: ', status);
-                    break;
+            data.email = data.email.trim();
+            
+            //Удаление отрисованных ошибок
+            for (const key in data) {
+                if (document.getElementById(key + 'Error') !== null) {
+                    document.getElementById(key + 'Error').remove();
                 }
             }
+
+            if (!this.validate(data)) {
+                return;
+            }
+
+            const r = new Req();
+            const [username, email, password, anotherPassword] = Array.from(data);
```

Comment:
```suggestion
            const [username, email, password] = data;
```
anotherPassword не используется и не очень понимаю зачем писать лишние буквы Array.from, поясни